tests/test_pulp_util.py

```diff
@@ -197,3 +198,69 @@ def test_get_pulp_instance():
     _, workflow = prepare(testfile)
     handler = PulpHandler(workflow, pulp_registry_name, log)
     assert handler.get_pulp_instance() == pulp_registry_name
+
+
+@pytest.mark.skipif(dockpulp is None,
+                    reason='dockpulp module not available')
+@pytest.mark.parametrize(("auto_publish"), [
+    (True),
+    (False),
+])
+@pytest.mark.parametrize(("fail"), [
+    (True),
+    (False)
+])
+def test_ensure_repos(auto_publish, fail):
+    dist_data = [
+          {
+            "repo_id": "redhat-myproject-hello-world",
+            "auto_publish": auto_publish,
+          }
+        ]
+    mock_get_data = [
+      {
+        "id": "redhat-myproject-hello-world",
+        "scratchpad": {
+          "tags": [
+            {
+              "image_id": "81c26020120f56f58217eefdbddaad37e0c47d1404e3323c6a7d2648325b08f1",
+              "tag": "none-80435-20170705193414"
+            }
+          ]
+        },
+        "display_name": "redhat-myproject-hello-world",
+        "description": "No description",
+      }
+    ]
+    data_with_dist = copy.deepcopy(mock_get_data)
+    data_with_dist[0]["distributors"] = dist_data
+
+    log = logging.getLogger("tests.test_pulp_util")
+    pulp_registry_name = 'registry.example.com'
+    testfile = 'foo'
+
+    _, workflow = prepare(testfile)
+    handler = PulpHandler(workflow, pulp_registry_name, log)
+    handler.create_dockpulp_and_repos([])
+
+    if fail:
+        (flexmock(dockpulp.Pulp)
+         .should_receive('getRepos')
+         .with_args(['redhat-myproject-hello-world'], fields=['id'], distributors=True)
+         .and_raise(TypeError))
+        (flexmock(dockpulp.Pulp)
+         .should_receive('getRepos')
+         .with_args(['redhat-myproject-hello-world'], fields=['id'])
+         .and_return(mock_get_data))
+    else:
+        (flexmock(dockpulp.Pulp)
+         .should_receive('getRepos')
+         .with_args(['redhat-myproject-hello-world'], fields=['id'], distributors=True)
+         .and_return(data_with_dist))
+
+    if auto_publish and not fail:
```

Comment:
Let's add an else with `.never()` in the expectation to ensure `updateRepo` is not called.